### PR TITLE
Update go to use 1.18.2

### PR DESCRIPTION
**Pull request by cpanato**
- update go in the actions and use the recommended way to get the latest
- update go base image to 1.18.2

### CODE REVIEW

File: .github/workflows/jsonnet.yml [hunk 1]
```diff
@@ -23,7 +23,8 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: ^1.16
+          go-version: '1.16'
```
Comment (cpanato) [inline .github/workflows/jsonnet.yml:26]: this can be upgraded to 1.18?

File: .github/workflows/release-dry-run.yml [hunk 2]
```diff
@@ -48,10 +48,11 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: ^1.16
+          go-version: '1.18'
```
Comment (cpanato) [inline .github/workflows/release-dry-run.yml:51]: I think we want  go1.18, but not 100% sure

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -22,7 +22,8 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: ^1.16
+          go-version: '1.18'
```
Comment (cpanato) [inline .github/workflows/release.yml:25]: here as well: I think we want  go1.18, but not 100% sure

File: .github/workflows/jsonnet.yml (hunk 1 above)
Comment (brancz) [reply .github/workflows/jsonnet.yml:26]: Yes!

File: .github/workflows/release-dry-run.yml (hunk 2 above)
Comment (brancz) [reply .github/workflows/release-dry-run.yml:51]: 100%

File: .github/workflows/release.yml (hunk 3 above)
Comment (brancz) [reply .github/workflows/release.yml:25]: lgtm

**Comment by cpanato** [comment]
found other places to update as well

**Comment by cpanato** [comment]
the nodejs is `node:17.7.1-alpine3.15 ` for `scripts/make-containers.sh` this is okay? or should we bump to a newer?

**Comment by brancz** [comment]
we can bump it!

**Comment by cpanato** [comment]
> we can bump it!

the node will do in a follow up PR

**Comment by kakkoyun** [comment]
@cpanato, I think you need to run `make proto/all` to update the buf.lock file. The CI fails because od that.

**Comment by cpanato** [comment]
@kakkoyun can you approve the run? thanks!

**Comment by cpanato** [comment]
humm, will check the golangci lint errors

**Comment by cpanato** [comment]
i will check the golangci-lint update later, dont know why that is failing

@kakkoyun PTAL, thanks for your review

**Review by kakkoyun — APPROVED** [review]
LGTM

**Pull request merged**